server/src/main/java/org/elasticsearch/snapshots/SnapshotsService.java

```diff
@@ -2387,6 +2387,12 @@ private void deleteSnapshotsFromRepository(
             assert currentlyFinalizing.contains(deleteEntry.repository());
             final List<SnapshotId> snapshotIds = deleteEntry.getSnapshots();
             assert deleteEntry.state() == SnapshotDeletionsInProgress.State.STARTED : "incorrect state for entry [" + deleteEntry + "]";
+            if (snapshotIds.isEmpty()) {
```

Comment:
I think this is fine as an optimization but we could do even better IMO. Couldn't we just resolve the listener on deletes that become empty on a CS update and remove them from the CS right away, thus saving us another CS update?

Reply:
I agree and I looked at it but that was less straightforward for me to make such a change (it needs to be done at several places) for some edge case situation like this. I'd go with this change and add some comments in the code about possible optimization. WDYT?